jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSourceTransformer.java

```diff
@@ -17,6 +17,17 @@
 
 import org.eclipse.jetty.io.Content;
 
+/**
+ * An abstract transformer of {@link Content.Chunk}s from a {@link Content.Source} as a {@link Content.Source}.
+ * <p>
+ * This abstract {@link Content.Source} wraps another {@link Content.Source} and implementors need only to provide
+ * the {@link #transform(Content.Chunk)} method, which is used to transform {@link Content.Chunk} read from the
+ * wrapped source.
+ * <p>
+ * The {@link #demand(Runnable)} conversation is passed directly to the wrapped {@link Content.Source}, which means
+ * that transformations that may fully consume bytes read can result in a null return from {@link Content.Source#read()}
+ * even after a callback to the demand {@link Runnable}.
```

Comment:
This last paragraph is (should?) already be covered by the fact that `demand()` may lead to spurious callbacks. I'd either remove this paragraph or link to the one that speaks about `demand()`'s spurious behavior.